packages/store/src/types.js

```diff
@@ -612,6 +612,11 @@
 /** @typedef {{ klass: 'methodGuard', callKind: 'sync' | 'async', returnGuard: unknown }} MethodGuard */
 /** @typedef {any} ArgGuard */
 
+// a parameter list like foo(a, b, c = d, …e) => f should be guarded by
+// something like
+// foo: M.call(AShape, BShape).optional(CShape).rest(EShape).returns(FShape)
+// optional is for optional (=) params. rest is for … (varargs) params
```

Comment:
The text looks good. Go ahead and put it into the `MethodGuardMaker` type doc-comment above, which should be defined as something with `optional`, `rest`, and `returns` methods.